src/client/pps.go

```diff
@@ -858,7 +858,14 @@ func (c APIClient) GetKubeEvents(since time.Duration) ([]*pps.LokiLogMessage, er
 	if err != nil {
 		return nil, grpcutil.ScrubGRPC(err)
 	}
-	return grpcutil.Collect[*pps.LokiLogMessage](client, 1000)
+	var resp []*pps.LokiLogMessage
+	if err := grpcutil.ForEach[*pps.LokiLogMessage](client, func(msg *pps.LokiLogMessage) error {
```

Comment:
BTW, are you sure that Go can't infer the type in `[*pps.LokiLogMessage]` from the type of the callback passed in?  Worth a try if you haven't already.

Reply:
hmm, will try this